runners/spark/src/main/java/org/apache/beam/runners/spark/TestSparkPipelineOptions.java

```diff
@@ -32,4 +36,22 @@
   boolean isForceStreaming();
   void setForceStreaming(boolean forceStreaming);
 
+  @Description("A hard-coded expected number of assertions for this test pipeline.")
+  @Nullable
+  Integer getExpectedAssertions();
```

Comment:
Some of the tests for the Spark runner test a recovery/resume from checkpoint, so while "countAssertions" might expect 1 assertion, it is fair that this assertion would only happen after recovery, so that first execution of the pipeline has 0 assertions, and the following that resumes from checkpoint has the actual expected, 1 assertion.
Usually I'd expect runners _not_ to test the underlying engine's features (such as recovery from checkpoint), but since Spark doesn't provide things like recovering metric values and the runner implements those we have to test them.